### PR TITLE
Enable io plugins to provide baddr ##io

### PR DESCRIPTION
**Pull request by condret**
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

I need this at work, but haven't tested if it actually works.

### CODE REVIEW

File: libr/main/radare2.c [hunk 1]
```diff
@@ -1108,6 +1110,12 @@ R_API int r_main_radare2(int argc, const char **argv) {
 						perms |= R_PERM_CREAT;
 						fh = r_core_file_open (r, pfile, perms, mapaddr);
 					}
+					if (!baddr_set) {
+						const ut64 io_plug_baddr = r_config_get_i (r->config, "bin.baddr");
+						if (io_plug_baddr) {
```
Comment (trufae) [inline libr/main/radare2.c:1115]: || io_plug_baddr != UT64_MAX

File: libr/main/radare2.c [hunk 2]
```diff
@@ -487,6 +488,7 @@ R_API int r_main_radare2(int argc, const char **argv) {
 			break;
 		case 'B':
 			baddr = r_num_math (r->num, opt.arg);
+			baddr_set = true;
```
Comment (trufae) [inline libr/main/radare2.c:491]: just check if baddr == UT64_MAX insteaed of having baddr_set var

File: libr/main/radare2.c [hunk 3]
```diff
@@ -1108,6 +1110,12 @@ R_API int r_main_radare2(int argc, const char **argv) {
 						perms |= R_PERM_CREAT;
 						fh = r_core_file_open (r, pfile, perms, mapaddr);
 					}
+					if (!baddr_set) {
```
Comment (trufae) [inline libr/main/radare2.c:1113]: == ut64max

**Review by trufae — CHANGES_REQUESTED** [review]
do u mean that r2 -B is not behaving properly in some situations? can you provide a way to reproduce this issue? is the io plugin setting bin.baddr via coreb maybe?

File: libr/main/radare2.c (hunk 2 above)
Comment (condret) [reply libr/main/radare2.c]: are you sure?

**Pull request closed without merging**